source/expressions.tex

```diff
@@ -1458,65 +1505,86 @@
 \end{bnf}
 
 \pnum
-The type denoted by a \grammarterm{decltype-specifier} in a
-\grammarterm{nested-name-specifier} shall be a class or enumeration
-type.
+\indextext{name!component}%
+The component names of a \grammarterm{qualified-id} are those of
+its \grammarterm{nested-name-specifier} and \grammarterm{unqualified-id}.
+The component names of a \grammarterm{nested-name-specifier} are
+its \grammarterm{identifier} (if any) and those of its
```

Comment:
No rewrite for this round; it seems sufficiently obvious "its" refers to the nested-name-specifier here,

Reply:
I parsed its as referring to the subject "The component names", so it threw me off.  Use a placeholder "S" for the /nested-name-specifier/ and replace "its" with "S" please?

Reply:
"Component names" are plural, so "its" cannot possibly refer to them. (It would be "their" otherwise.)
The sentence doesn't feel complicated enough to warrant meta-variables such as $S$.
Not changing; for @tkoeppe  to decide.

Reply:
I agree that this isn't ambiguous. I support avoiding ambiguity, but close reading is all that's needed in this case. I'm generally in favour of using pronouns only sparingly and only with care, but this use case seems reasonable, and the alternative doesn't seem to buy much.